### PR TITLE
Clarify section regarding additional board members

### PR DESCRIPTION
**Pull request by gheine**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,11 +24,20 @@ General Structure
...
   - The board reports to SVP Engineering.
 
+Additional Board Members
+------------------------
+
+  - The board should review its composition before each Architecture Council.
+  - In order to ensure broad technical and departmental representation, the board may appoint additional members at any point.
+  - Additional members are proposed by the Chair, after consultation with the respective department leads/directors.
+  - The board decides on the proposed members via majority vote.
+  - Additional board members are given a permanent seat on the board.
```
Comment (andreyk0) [inline README.md]: Weren't there some term limits in the past (which I've personally probably exceeded at this point) or have we changed the rules?

Comment (gheine) [inline README.md]: We got rid of that when we expanded the board to all PE & DE's. Further up it says: "The principal and distinguished members of Gilt Tech constitute the permanent members of the architecture board."

**Comment by arbjerg** [comment]
+1

File: README.md (hunk 1 above)
Comment (andreyk0) [inline README.md]: Right. NM

**Comment by andreyk0** [comment]
👍

**Comment by njprrogers** [comment]
👍

**Pull request merged**